### PR TITLE
Use XLA utility API to move data to CPU (Single TPU core)

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Use XLA utility API to move data to CPU

## Before submitting
- [] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [] Did you verify new and existing tests pass locally with your changes?
- [] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGMT !

File: pytorch_lightning/plugins/training_type/single_tpu.py [hunk 1]
```diff
@@ -54,13 +56,20 @@ def pre_dispatch(self) -> None:
         self.tpu_local_core_rank = xm.get_local_ordinal()
         self.tpu_global_core_rank = xm.get_ordinal()
 
-    def on_save(self, checkpoint: dict) -> dict:
```
Comment (carmocca) [inline pytorch_lightning/plugins/training_type/single_tpu.py:57]: Do we not want to rely on the training type API?

Comment (kaushikb11) [reply pytorch_lightning/plugins/training_type/single_tpu.py:57]: Wdym? `save_checkpoint` is part of the Accelerator API.

Comment (carmocca) [reply pytorch_lightning/plugins/training_type/single_tpu.py:57]: I mean that maybe

```python
        if _OMEGACONF_AVAILABLE:
            checkpoint = apply_to_collection(checkpoint, (DictConfig, ListConfig), OmegaConf.to_container)
```

and the `callback` key filter

should be kept under `on_save`

and we just need a hook to override `atomic_save`

Comment (kaushikb11) [reply pytorch_lightning/plugins/training_type/single_tpu.py:57]: Ahh, I see. So, `save_checkpoint` would have two hooks then (`on_save` and `save`)

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/single_tpu.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/plugins/training_type/single_tpu.py:57]: Yes, that's the idea.

Comment (kaushikb11) [reply pytorch_lightning/plugins/training_type/single_tpu.py:57]: @carmocca This wouldn't work for `tpu_spawn`. In `save_checkpoint`, it only saves the checkpoint for global rank zero. But for TPUs, there's a rendevous in `xm.save` method hence all processes/ranks need to reach that part.

File: pytorch_lightning/plugins/training_type/single_tpu.py [hunk 2]
```diff
@@ -54,13 +56,20 @@ def pre_dispatch(self) -> None:
...
-        """
-        Move XLA tensors to CPU before saving
-        Recommended on XLA Guide:
-        https://github.com/pytorch/xla/blob/master/API_GUIDE.md#saving-and-loading-xla-tensors
+    def save(self, state_dict: Dict, path: str) -> None:
+        xm.save(state_dict, path)
```
Comment (carmocca) [inline pytorch_lightning/plugins/training_type/single_tpu.py:60]: Is there a reason to put this line under a function call? Do you plan to re-use it?

Comment (kaushikb11) [reply pytorch_lightning/plugins/training_type/single_tpu.py:60]: Yup. Planning to explore one more XLA saving functionality.

**Review by carmocca — APPROVED** [review]
